### PR TITLE
AR Sample: View Hidden Infrastructure

### PR DESCRIPTION
**Pull request by ZackAllen**

### CODE REVIEW

File: src/Android/Xamarin.Android/ArcGISRuntime.Xamarin.Samples.Android.csproj [hunk 1]
```diff
@@ -256,6 +256,7 @@
     <None Include="Samples\Symbology\SimpleRenderers\SimpleRenderers.jpg" />
     <None Include="Samples\Symbology\UseDistanceCompositeSym\UseDistanceCompositeSym.jpg" />
     <None Include="Samples\Security\IntegratedWindowsAuth\readme.md" />
+    <None Include="Samples\Augmented reality\CollectDataAR\CollectDataAR.jpg" />
```
Comment (nCastle1) [inline src/Android/Xamarin.Android/ArcGISRuntime.Xamarin.Samples.Android.csproj]: Is there a `ViewHiddenInfrastructureAR.jpg` entry?

File: src/Android/Xamarin.Android/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                else
+                {
+                    _helpLabel.Text = string.Format("Pipe added {0:0.0}m above the surface", elevationOffset);
+                }
+
+                // Enable the view button once a pipe has been added to the graphics overlay.
+                _viewButton.Enabled = true;
+            }
+            catch (Exception)
+            {
```
Comment (nCastle1) [inline src/Android/Xamarin.Android/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs]: Might be worth showing these?

File: src/Android/Xamarin.Android/Samples/Augmented Reality/ViewHiddenInfrastructureAR/readme.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## Additional information
+
+This sample requires a device that is compatible with ARKit 1 on iOS or ARCore 1.8 on Android.
+
+Note that unlike other scene samples, a basemap isn't shown most of the time, because the real world provides the context. Only while calibrating is the basemap displayed at 50% opacity, to give the user a visual reference to compare to.
+
+You may notice that pipes you draw underground appear to float more than you would expect. That floating is a normal result of the parallax effect that looks unnatural because you're not used to being able to see underground/obscured objects. Compare the behavior of underground pipes with equivalent pipes drawn above the surface - the behavior is the same, but probably feels more natural above ground because you see similar scenes day-to-day (e.g. utility wires).
+
+**Real-scale AR** is one of three main patterns for working with geographic information in augmented reality. See  [Agumented reality]() in the guide for more information.
```
Comment (nCastle1) [inline src/Android/Xamarin.Android/Samples/Augmented Reality/ViewHiddenInfrastructureAR/readme.md]: Doc link: https://developers.arcgis.com/net/latest/android/guide/display-scenes-in-augmented-reality.htm

File: src/iOS/Xamarin.iOS/ArcGISRuntime.Xamarin.Samples.iOS.csproj [hunk 4]
```diff
@@ -684,7 +687,9 @@
   <ItemGroup>
     <EmbeddedResource Include="Resources\PictureMarkerSymbols\pin_blue.png" />
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup>
+    <BundleResource Include="Samples\Augmented reality\ViewHiddenInfrastructureAR\ViewHiddenInfrastructureAR.jpg" />
```
Comment (nCastle1) [inline src/iOS/Xamarin.iOS/ArcGISRuntime.Xamarin.Samples.iOS.csproj]: This should be a `None` include near the other JPGs

File: src/iOS/Xamarin.iOS/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                else
+                {
+                    _helpLabel.Text = string.Format("Pipe added {0:0.0}m above the surface", elevationOffset);
+                }
+
+                // Enable the view button once a pipe has been added to the graphics overlay.
+                _viewButton.Enabled = true;
+            }
+            catch (Exception)
+            {
```
Comment (nCastle1) [inline src/iOS/Xamarin.iOS/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs]: Maybe log this exception?

File: src/iOS/Xamarin.iOS/Samples/Augmented Reality/ViewHiddenInfrastructureAR/readme.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+## Additional information
+
+This sample requires a device that is compatible with ARKit 1 on iOS or ARCore 1.8 on Android.
+
+Note that unlike other scene samples, a basemap isn't shown most of the time, because the real world provides the context. Only while calibrating is the basemap displayed at 50% opacity, to give the user a visual reference to compare to.
+
+You may notice that pipes you draw underground appear to float more than you would expect. That floating is a normal result of the parallax effect that looks unnatural because you're not used to being able to see underground/obscured objects. Compare the behavior of underground pipes with equivalent pipes drawn above the surface - the behavior is the same, but probably feels more natural above ground because you see similar scenes day-to-day (e.g. utility wires).
+
+**Real-scale AR** is one of three main patterns for working with geographic information in augmented reality. See  [Agumented reality]() in the guide for more information.
```
Comment (nCastle1) [inline src/iOS/Xamarin.iOS/Samples/Augmented Reality/ViewHiddenInfrastructureAR/readme.md]: Doc link: https://developers.arcgis.com/net/latest/ios/guide/display-scenes-in-augmented-reality.htm

File: src/iOS/Xamarin.iOS/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs (hunk 5 above)
Comment (ZackAllen) [reply src/iOS/Xamarin.iOS/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs]: I added a write to the console. Exception will only occur if something happens in GetElevationAsync. Add sketch can't be called until the elevation surface is already loaded.

File: src/Android/Xamarin.Android/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+            // Start the AR viewer activity.
+            Intent myIntent = new Intent(this, typeof(PipeViewer));
+            StartActivity(myIntent);
+        }
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Sketch on map";
```
Comment (nCastle1) [inline src/Android/Xamarin.Android/Samples/Augmented Reality/ViewHiddenInfrastructureAR/PipePlacer.cs]: I think this title needs to be updated

**Pull request merged**